### PR TITLE
[FLASK] Create E2E Test for BIP-32 test-snap

### PR DESCRIPTION
**Pull request by bowensanders**
This PR does a few things:

- creates a new E2E test for the bip-32 test snap
- changes the way that permissions checkboxes are id'd (ui/pages/permissions-connect/flask/utils.js)
- very small fix in the related bip-44 E2E test to use checkbox properly
- increments version of test-snaps that is used
- fixes issues with e2es running on the latest version of test-snaps

### CODE REVIEW

File: test/e2e/snaps/test-snap-bip32.spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+const { strict: assert } = require('assert');
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-bip32.spec.js:1]: Looks to be some linting errors. Should be fixable with `yarn lint:fix`?

File: test/e2e/snaps/test-snap-bip32.spec.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+const { check } = require('yargs');
+const { withFixtures } = require('../helpers');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+describe('Test Snap update', function () {
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-bip32.spec.js:6]:
```suggestion
describe('Test Snap bip-32', function () {
```

File: test/e2e/snaps/test-snap-bip32.spec.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  it('can install an old and then updated version', async function () {
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-bip32.spec.js:7]: This is also incorrect, not sure what it should be. 

We may have to change some of the snap's logic as well. See Slack.

File: ui/pages/permissions-connect/flask/util.js [hunk 4]
```diff
@@ -17,7 +17,9 @@ export function getSnapInstallWarnings(permissions, targetSubjectMetadata, t) {
   return [
     ...flatMap(bip32EntropyPermissions, (permission, i) =>
       permission.caveats[0].value.map(({ path, curve }) => ({
-        id: `key-access-bip32-${path.join('/')}-${curve}-${i}`,
+        id: `key-access-bip32-${path
+          .join('-')
+          .replace(/\'/g, 'h')}-${curve}-${i}`,
```
Comment (FrederikBolding) [inline ui/pages/permissions-connect/flask/util.js:22]:
```suggestion
          .replace(/'/gu, 'h')}-${curve}-${i}`,
```
Escaping is not needed and the linter wants `/gu`

**Review by FrederikBolding — DISMISSED** [review]

File: test/e2e/snaps/test-snap-bip32.spec.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          await publicKeyResult.getText(),
+          'Public key: "043e98d696ae15caef75fa8dd204a7c5c08d1272b2218ba3c20feeb4c691eec366606ece56791c361a2320e7fad8bcbb130f66d51c591fc39767ab2856e93f8dfb"',
+        );
+
+        // wait then run compressed public key test
+        await driver.delay(1000);
+        await driver.clickElement('#sendBip32CompressedPublicKey');
+        // check result
+        await driver.delay(1000);
+        const CompressedPublicKeyResult = await driver.findElement(
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-bip32.spec.js:125]: Please use camelCase

File: test/e2e/snaps/test-snap-bip32.spec.js (hunk 2 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-bip32.spec.js]: fixed

File: test/e2e/snaps/test-snap-bip32.spec.js (hunk 1 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-bip32.spec.js:1]: I think i have dispatched all the linting errors. There's still a warning but I don't think that will fail and it isn't in anything i touched with this PR

File: test/e2e/snaps/test-snap-bip32.spec.js (hunk 3 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-bip32.spec.js]: I fixed that. It was an over-copy from a previous test and I have remedied it

**Review by GuillaumeRx — DISMISSED** [review]

File: test/e2e/snaps/test-snap-bip-32.spec.js [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+describe('Test Snap bip-32', function () {
+  it('tests various functions of bip-32', async function () {
+    const ganacheOptions = {
+      accounts: [
+        {
+          secretKey:
+            '0x7C9529A67102755B7E6102D6D950AC5D5863C98713805CEC576B945B15B71EAC',
+          balance: 25000000000000000000,
```
Comment (Mrtenz) [inline test/e2e/snaps/test-snap-bip-32.spec.js:12]: Can this be a string or `bigint`? It's not a safe integer, and may result in unexpected behaviour.

File: ui/pages/permissions-connect/flask/util.js [hunk 7]
```diff
@@ -17,7 +17,9 @@ export function getSnapInstallWarnings(permissions, targetSubjectMetadata, t) {
   return [
     ...flatMap(bip32EntropyPermissions, (permission, i) =>
       permission.caveats[0].value.map(({ path, curve }) => ({
-        id: `key-access-bip32-${path.join('/')}-${curve}-${i}`,
+        id: `key-access-bip32-${path
+          .join('-')
+          .replace(/'/gu, 'h')}-${curve}-${i}`,
```
Comment (Mrtenz) [inline ui/pages/permissions-connect/flask/util.js:22]:
```suggestion
          .replaceAll(`'`, 'h')}-${curve}-${i}`,
```

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: ui/pages/permissions-connect/flask/util.js (hunk 7 above)
Comment (FrederikBolding) [reply ui/pages/permissions-connect/flask/util.js:22]: https://caniuse.com/?search=replaceAll - we cant use that.

File: test/e2e/snaps/test-snap-bip-32.spec.js (hunk 6 above)
Comment (FrederikBolding) [reply test/e2e/snaps/test-snap-bip-32.spec.js:12]: This is used in all the other E2Es, we can do a separate PR to fix this if you don't mind?

**Review by Mrtenz — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**